# **MÓDULO 26 - Projeto Final do Aprofundamento de Analytics**

Bem-vindos ao Projeto de Dashboard de E-commerce! Este projeto é uma oportunidade para vocês aplicarem habilidades essenciais de análise de dados em um cenário prático e realista. Vocês irão trabalhar com um conjunto de dados de transações de clientes de uma loja virtual, distribuídos em duas tabelas distintas. O objetivo final é construir um dashboard interativo que facilite a visualização e análise das informações relevantes do e-commerce, utilizando ferramentas como Looker Studio ou Power BI.

**Objetivo do Projeto:**

Tratamento de Dados: Realizar a junção (JOIN) de duas tabelas utilizando SQL para consolidar as informações.
Análise de Dados: Exportar os dados resultantes para um arquivo CSV.
Visualização de Dados: Desenvolver um dashboard interativo e informativo para visualização das principais métricas e insights do e-commerce.

**Tabelas Disponibilizadas:**

**Tabela de Transações:** Contém os registros de transações realizadas pelos clientes, incluindo detalhes como ID da transação, valor e outros.


**Tabela de Dados Pessoais:** Contém as informações pessoais dos clientes, como ID do cliente, nome, genero, cidade, etc.

**Chave de Ligação:** As tabelas se relacionam através da coluna ID_CLIENT, que é a chave identificadora dos clientes.

# Etapas do Projeto:

1. Realizar um JOIN SQL nas duas tabelas, unificando as informações através da coluna ID_CLIENT. Você deve justificar a escolha do JOIN (Inner/ Left/ Right ou Full).

2. Exportar os dados consolidados resultantes do JOIN para um arquivo CSV.

3. Utilizar Looker Studio ou Power BI para importar o arquivo CSV.

4. Criar visualizações interativas que apresentem métricas importantes, como total de vendas, número de transações, distribuição geográfica dos clientes, perfil demográfico dos clientes, entre outros.

Abaixo temos a configuração do ambiente SQL:

In [8]:
import sqlite3
import pandas as pd

In [9]:
df_transacoes = pd.read_csv("TB_TRANSACOES_PROJETO_ECOMM.csv", delimiter=';')
df_clientes = pd.read_csv("TB_CLIENTES_PROJETO_ECOMM.csv", delimiter=';')

In [10]:
conn = sqlite3.connect('projeto.db')
# Carregar o DataFrame no banco de dados SQLite - criando tb_transacoes e tb_clientes
df_transacoes.to_sql('tb_transacoes', conn, index=False, if_exists='replace')
df_clientes.to_sql('tb_clientes', conn, index=False, if_exists='replace')

175

In [11]:
# Função para executar consultas SQL e retornar o resultado como um DataFrame
def run_query(query):
    return pd.read_sql_query(query, conn)

In [12]:
df_transacoes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   id_client  367 non-null    int64 
 1   Category   367 non-null    object
 2   Price      367 non-null    object
 3   Card Type  367 non-null    object
dtypes: int64(1), object(3)
memory usage: 11.6+ KB


In [18]:
query = """
SELECT
COUNT(DISTINCT(id_client))
FROM tb_transacoes
"""
result_df = run_query(query)
print(result_df)

   COUNT(DISTINCT(id_client))
0                         241


In [14]:
df_clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   state_name  175 non-null    object
 1   First_name  175 non-null    object
 2   Gender      175 non-null    object
 3   Job_Title   175 non-null    object
 4   Id_client   175 non-null    int64 
dtypes: int64(1), object(4)
memory usage: 7.0+ KB


In [19]:
query = """
SELECT
COUNT(DISTINCT(id_client))
FROM tb_clientes
"""
result_df = run_query(query)
print(result_df)

   COUNT(DISTINCT(id_client))
0                         175


In [15]:
print(df_transacoes.head())
print(df_clientes.head())

   id_client     Category   Price   Card Type
0         37  Electronics   72,93  mastercard
1         38      Jewelry  121,89  mastercard
2         39         Baby    64,3  mastercard
3         40     Outdoors    9,48  mastercard
4          5     Outdoors   61,95  mastercard
  state_name First_name  Gender                     Job_Title  Id_client
0         TX    Domingo    Male  Structural Analysis Engineer          1
1         MI    Russell    Male            Speech Pathologist          2
2         AL     Kimble    Male           Account Coordinator          3
3         IL   Barnabas    Male               General Manager          4
4         MN     Tanney  Female                  VP Marketing          5


In [16]:
query = """
SELECT
*
FROM
tb_transacoes AS A
INNER JOIN tb_clientes AS B
ON A.Id_client = B.Id_client
ORDER BY A.Id_client ASC
"""
result_df = run_query(query)
print(result_df)

     id_client    Category   Price   Card Type state_name First_name   Gender  \
0            1    Outdoors   16,97  mastercard         TX    Domingo     Male   
1            2     Grocery  143,39  mastercard         MI    Russell     Male   
2            3       Music   37,64  mastercard         AL     Kimble     Male   
3            4     Jewelry    8,51  mastercard         IL   Barnabas     Male   
4            5    Outdoors   61,95  mastercard         MN     Tanney   Female   
..         ...         ...     ...         ...        ...        ...      ...   
291        171       Books   35,85  mastercard         UT  Fairleigh     Male   
292        172  Automotive  130,08  mastercard         DC    Lambert  Agender   
293        173        Kids   28,38  mastercard         NY      Jacob     Male   
294        174      Sports   36,35  mastercard         OK        Pip     Male   
295        175        Kids  141,71  mastercard         OR     Murvyn   Female   

                        Job

In [22]:
query = """
SELECT
COUNT(DISTINCT(A.Id_client))
FROM tb_transacoes AS A
INNER JOIN tb_clientes AS B
ON A.Id_client = B.Id_client
"""
print(run_query(query))

   COUNT(DISTINCT(A.Id_client))
0                           170


Justifique a escolha do JOIN: Considerando que há 241 clientes diferentes em 367 transações e 175 clientes cadastrados no bancos de dados, escolhemos o INNER JOIN pois desta forma, teriamos quase todas as transações dos clientes cadastrados. Isto é, temos as 296 transações de 170 dos 175 clientes cadastrados.

Exportando o arquivo como CSV:

In [17]:
result_df.to_csv('dados_ecommerce_final.csv', index=False)

**Dicas para o projeto:**
- Se atente que, como o mesmo cliente realiza mais de 1 transação quando você for trazer alguma métrica relacionada a dados do cliente terá que utilizar o distinct para criar essas métricas no dashboard, se não acabará tendo os dados repetidos.

- Análise sua tabela, entenda a dimensão dos dados, no excel, antes de enviar para o Powerbi ou Looker Studio.

- Tente montar preveamente um roteiro de quais métricas e visualizações irá colocar no dashboard, isso tornará seu processo mais rápido.

- Qualquer dificuldade para subir sua base para as ferramentas de visualização envie a base e o erro encontrado para que os tutores possam te ajudar.